rero_ils/modules/selfcheck/api.py

```diff
@@ -175,7 +183,76 @@ def patron_information(barcode, **kwargs):
                     loan = Loan.get_record_by_pid(transaction.loan_pid)
                     item = Item.get_record_by_pid(loan.item_pid)
                     patron_account_information.get('fine_items', []).append(
-                        item.get('barcode')
+                        item.get('pid')
                     )
-
         return patron_account_information
+
+
+def item_information(patron_barcode, item_pid, **kwargs):
+    """Get item information handler.
+
+    get item information according 'item_identifier' from selfcheck user.
+    :param barcode: barcode of the patronr.
+    :param item_pid: item identifier.
+    :return: The SelfcheckItemInformation object.
+    """
+    # check if invenio_sip2 module is present
+    if check_sip2_module():
+        from invenio_sip2.models import SelfcheckFeeType, \
+            SelfcheckItemInformation, SelfcheckSecurityMarkerType
+        patron = Patron.get_patron_by_barcode(patron_barcode)
+        item = Item.get_record_by_pid(item_pid)
+        document = Document.get_record_by_pid(item.document_pid)
+        location = item.get_location()
+        language = kwargs.get('language')
+        with current_app.test_request_context() as ctx:
+            ctx.babel_locale = language
+            item_information = SelfcheckItemInformation(
+                item_id=item.get('barcode'),
+                title_id=title_format_text_head(document.get('title')),
+                circulation_status=map_item_circulation_status(item.status),
+                fee_type=SelfcheckFeeType.OTHER,
+                security_marker=SelfcheckSecurityMarkerType.OTHER
+            )
+            item_information['media_type'] = map_media_type(
+                document.get('type')
+            )
+            item_information['hold_queue_length'] = item.number_of_requests()
+            item_information['owner'] = location.get_library().get('name')
+            item_information['permanent_location'] = location.get('name')
+            item_information['current_location'] = \
+                item.get_last_location().get('name')
+            # get loan for item
+            filter_states = [
+                LoanState.PENDING,
+                LoanState.ITEM_AT_DESK,
+                LoanState.ITEM_IN_TRANSIT_FOR_PICKUP,
+                LoanState.ITEM_ON_LOAN
```

Comment:
do you need to filter by `LoanState.ITEM_AT_DESK` and `LoanState.ITEM_IN_TRANSIT_FOR_PICKUP`? 
you may remove them if not used

Reply:
You are right. We doesn't need that.